Support for JSONC

IzhakJakov opened the pull request:
## Description from issue https://github.com/vim/vim/issues/8499

I see more and more tools using jsonc format for their configuration files.  I believe this trend will only increase and VIM should consider supporting these formats

### [jsonc](https://komkom.github.io)
Basically a JSON with C-style line (`//`) and block (`/**/`) comments.
Exiting vim plugin: https://github.com/kevinoid/vim-jsonc

### Steps to reproduce using `vim -u NONE`

```
$ vim -u NONE sample.jsonc
<opened as a plain text>
```

### Actual behaviour
opens up as a plain text

### Expected behaviour
Opened as a jsonc file:
- proper syntax highlighting
- proper error messages
- proper indentation and formatting

Commit:
Support for JSONC

dkearns commented on runtime/syntax/jsonc.vim:
```diff
@@ -0,0 +1,45 @@
+endif
+
+" Based on vim-json syntax
+runtime syntax/json.vim
```
You might want to use `runtime!` (the bang version) so that all JSON syntax files in the runtime path are sourced.

IzhakJakov replied:
Done.

Commit:
Changes to be committed:
	new file:   runtime/ftplugin/jsonc.vim
	modified:   runtime/filetype.vim
	modified:   runtime/syntax/jsonc.vim
	modified:   src/testdir/test_filetype.vim

Commit:
"runtime ..." -> "runtime! ..."
